src/async_connection.rs

```diff
@@ -0,0 +1,543 @@
+use std::io;
```

Comment:
is this another (older? version) of the system in `src/async`? it seems weird and self contained